express_relay/sdk/js/src/index.ts

```diff
@@ -303,52 +343,106 @@ export class Client {
    * Creates a signed bid for an opportunity
    * @param opportunity Opportunity to bid on
    * @param bidParams Bid amount and valid until timestamp
+   * @param opportunityAdapterConfig Opportunity adapter config
    * @param privateKey Private key to sign the bid with
+   * @returns Signed opportunity bid
    */
   async signOpportunityBid(
     opportunity: Opportunity,
     bidParams: BidParams,
+    opportunityAdapterConfig: OpportunityAdapterConfig,
     privateKey: Hex
   ): Promise<OpportunityBid> {
     const types = {
-      ExecutionParams: [
-        { name: "sellTokens", type: "TokenAmount[]" },
+      PermitBatchWitnessTransferFrom: [
+        { name: "permitted", type: "TokenPermissions[]" },
+        { name: "spender", type: "address" },
+        { name: "nonce", type: "uint256" },
+        { name: "deadline", type: "uint256" },
+        { name: "witness", type: "OpportunityWitness" },
+      ],
+      OpportunityWitness: [
         { name: "buyTokens", type: "TokenAmount[]" },
         { name: "executor", type: "address" },
         { name: "targetContract", type: "address" },
         { name: "targetCalldata", type: "bytes" },
         { name: "targetCallValue", type: "uint256" },
-        { name: "validUntil", type: "uint256" },
         { name: "bidAmount", type: "uint256" },
       ],
       TokenAmount: [
         { name: "token", type: "address" },
         { name: "amount", type: "uint256" },
       ],
+      TokenPermissions: [
+        { name: "token", type: "address" },
+        { name: "amount", type: "uint256" },
+      ],
     };
 
     const account = privateKeyToAccount(privateKey);
     const signature = await signTypedData({
       privateKey,
       domain: {
-        ...opportunity.eip712Domain,
-        chainId: Number(opportunity.eip712Domain.chainId),
+        name: "Permit2",
+        verifyingContract: checkAddress(opportunityAdapterConfig.permit2),
+        chainId: opportunityAdapterConfig.chain_id,
       },
       types,
-      primaryType: "ExecutionParams",
+      primaryType: "PermitBatchWitnessTransferFrom",
       message: {
-        sellTokens: opportunity.sellTokens,
-        buyTokens: opportunity.buyTokens,
-        executor: account.address,
-        targetContract: opportunity.targetContract,
-        targetCalldata: opportunity.targetCalldata,
-        targetCallValue: opportunity.targetCallValue,
-        validUntil: bidParams.validUntil,
-        bidAmount: bidParams.amount,
+        permitted: getPermittedTokens(
+          opportunity.sellTokens,
+          bidParams.amount,
+          opportunity.targetCallValue,
+          checkAddress(opportunityAdapterConfig.weth)
+        ),
+        spender: this.computeCreate2Address(
+          account.address,
+          checkAddress(opportunityAdapterConfig.opportunity_adapter_factory),
+          checkHex(
+            opportunityAdapterConfig.opportunity_adapter_init_bytecode_hash
+          )
+        ),
+        nonce: bidParams.nonce,
```

Comment:
nonce should have a default random value. Otherwise, we should document somewhere on what this value should be